app.js

```diff
@@ -19,10 +19,10 @@ Array.prototype._forEach = function(callbackFunction) {
 	}
 };
 
-const _isEqual = (array1, array2) => {
+Array.prototype._isEqual = function(array2) {
 	let result = [];
-	for (let i = 0; i < array1.length; i++) {
```

Comment:
antes del for debes agregar el siguiente if, de lo contrario cuando el `this` es un array vacío, y el array2 tiene algo, devuelve true

if (this.length===0){
	    return false;
	}